cmd/ssh.go

```diff
@@ -647,8 +647,19 @@ func startWorkspaceCredentialServer(ctx context.Context, client *ssh.Client, use
 	args = append(args, "--runner")
 	command = fmt.Sprintf("%s %s", command, strings.Join(args, " "))
 
-	if err := devssh.Run(ctx, client, command, stdin, stdout, writer, nil); err != nil {
-		return fmt.Errorf("run credentials server: %w", err)
+	for {
+		select {
+		case <-ctx.Done():
+			log.Infof("Context canceled, stopping credentials server: %v", ctx.Err())
+			return nil
+		default:
+			if err := devssh.Run(ctx, client, command, stdin, stdout, writer, nil); err != nil {
```

Comment:
I think we need a sleep in case of repeated failures

Reply:
ah a back off mechanism, yeah good point I think we even talked about it. 👀

Reply:
You could use `retry.OnError` from `"k8s.io/client-go/util/retry"` if you want something out of the box